lib/std/Build/Module.zig

```diff
@@ -284,6 +285,7 @@ pub fn init(
                 .owner = owner,
                 .root_source_file = if (options.root_source_file) |lp| lp.dupe(owner) else null,
                 .import_table = .{},
+                .allow_deprecated = owner.graph.allow_deprecated orelse !owner.is_root,
```

Comment:
In this version of the code modules track individually whether they're allowed to use deprecated code or not, giving to users the option of overriding the default behavior. See later comments for an alternative way of approaching this.